internal/runners/artifacts/artifacts.go

```diff
@@ -382,5 +384,25 @@ func getBuildPlan(
 		return nil, errs.New("Unhandled case")
 	}
 
+	// Note: the Platform does not raise an error when requesting a commit ID that does not exist in
+	// a given project, so we have verify existence client-side. See DS-1705 (yes, DS, not DX).
+	var owner, name, nsString string
+	if pj != nil {
+		owner = pj.Owner()
+		name = pj.Name()
+		nsString = pj.NamespaceString()
+	} else {
+		owner = namespace.Owner
+		name = namespace.Project
+		nsString = namespace.String()
+	}
+	_, err = model.GetCommitWithinProjectHistory(commit.CommitID, owner, name, auth)
+	if err != nil {
+		if err == model.ErrCommitNotInHistory {
+			return nil, &errCommitDoesNotExistInProject{nsString, commit.CommitID.String()}
```

Comment:
```suggestion
			return nil, errs.Pack(err, &errCommitDoesNotExistInProject{nsString, commit.CommitID.String()})
```

Even though we've interpreted it, we should still bubble up the underlying error. It may come in useful in the future for debugging purposes.